subtests/docker_cli/build/build.py

```diff
@@ -302,13 +347,40 @@ class git_path(build_base):
     pass
 
 
-class bad(build_base):
+class bad(build_base):  # negative test
+
+    def postprocess_result(self, build_def):
+        self.postprocess_negative(build_def)
+        self.postprocess_created_images(build_def)
+        self.postprocess_created_containers(build_def)
+        self.postprocess_container_count(build_def)
+
+
+class bad_quiet(bad):  # negative test
     pass
 
 
-class bad_quiet(build_base):
+class bad_force_rm(bad):  # negative test
     pass
 
 
-class bad_force_rm(build_base):
+class rm_false(build_base):
     pass
+
+
+class cache(build_base):
+
+    def make_builds(self, source):
+        first = source.deepcopy()
+        second = source.deepcopy()
```

Comment:
Nope, this is a typo on my part, you are correct ``copy.deepcopy()`` should be used.